UDx-container/Dockerfile_DEB

```diff
@@ -11,83 +11,240 @@
+FROM ubuntu:${os_version} 
 
+ARG vertica_db_user="dbadmin"
+ARG vertica_db_group="verticadba"
+
+ARG dbadmin_gid=1000
+ARG dbadmin_id=1000
+
+# useful to keep dpkg from prompting us for information
+ENV DEBIAN_FRONTEND noninteractive
+ENV TERM 1
+
+ENV VERTICA_OPT_DIR="/opt/vertica"
+ENV VERTICA_OPT_BIN="${VERTICA_OPT_DIR}/bin"
+ENV VERTICA_HOME_DIR="/home/$vertica_db_user"
+ENV VERTICA_VOLUME_DIR="/data"
+ENV VERTICA_DATA_DIR="${VERTICA_VOLUME_DIR}/vertica"
+ENV VERTICA_DB_NAME="vsdk"
+ENV PYTHON_EGG_CACHE="/tmp/.python-eggs"
+
+COPY --from=builder $VERTICA_OPT_DIR $VERTICA_OPT_DIR
+COPY --from=builder $VERTICA_DATA_DIR $VERTICA_DATA_DIR
+
+# this is a script which removes unnecessary stuff from the
+# container image
+COPY ./tools/init.d.functions /etc/rc.d/init.d/functions
+
+ADD ./tools/package-checksum-patcher.py /tmp/
+ADD ./tools/init.d.functions /etc/rc.d/init.d/functions
+
+# Install prerequisites, create Vertica data dir, document fix
+# timezone issue, set privileges, limits and generate locales  
+RUN set -x \
+ # COPY doesn't preserve the mode of the top-level
+ && chmod g+w ${VERTICA_OPT_DIR} \
+ && /usr/sbin/groupadd -r -g $dbadmin_gid $vertica_db_group \
+ && /usr/sbin/useradd -r \
+                      -m \
+                      -s /bin/bash \
+                      -g $vertica_db_group \
+                      # add dbadmin to root to work around COPY
+                      # not preserving owner and group prior to
+                      # docker version 19.03
+                      -G root \
+                      -u $dbadmin_id \
+                      $vertica_db_user \
+ && su - $vertica_db_user -c "mkdir $PYTHON_EGG_CACHE" \
+# Install prerequisites, create Vertica data dir, set privileges, document
+# fix timezone issue and generate locales in case of centOS 8 
+ && apt-get update -y \
+ && apt-get upgrade --no-install-recommends -y \
+ && apt-get install -y \
+            autotools-dev \
+            bash \
+            binutils \
+            build-essential \
+            cmake \
+            dialog \
+            doxygen \
+            expat \
+            flex \
+            # debian g++
+            g++ \
+            gcc \
+            gdb \
+            gettext \
+            iproute2 \
+            libacl1-dev \
+            libboost-all-dev \
+            libbz2-dev \
+            # centos: glibc-devel
+            libc-dev \
+            libcurl4-openssl-dev \
+            libgomp1 \
+            libicu-dev \
+            libkeyutils1 \
+            libreadline-dev \
+            libtool \
+            locales \
+            make \
+            nasm \
+            ntp \
+            openssl \
+            procps \
+            sudo \
+            sysstat \
+            tar \
+            vim \
+ && chsh -s /bin/bash root \
+ && /bin/echo "en_US ISO-8859-1" > /etc/locale.gen \
+ && /bin/echo "en_US.UTF-8 UTF-8" >> /etc/locale.gen \
+ && /usr/sbin/locale-gen \
```

Comment:
Dockerfiles have lots of quirks.  This is a two-stage dockerfile.  It opens with the first pass, creating a container it will call "builder":
`FROM ubuntu:${os_version} as builder`
which builds a huge container with lots of "layers", and layers take up space.
Then we start over with a fresh container:
`FROM ubuntu:${os_version}`
following up with:
```COPY --from=builder $VERTICA_OPT_DIR $VERTICA_OPT_DIR
COPY --from=builder $VERTICA_DATA_DIR $VERTICA_DATA_DIR
```
This copies the filesystem results of the earlier pass into this new container, without copying all the intermediate layers.
So we have to duplicate some work that didn't get copied.